flang/lib/Lower/OpenMP.cpp

```diff
@@ -266,13 +261,175 @@ genOMP(Fortran::lower::AbstractConverter &converter,
+          lastPrivateClause->v;
+      genObjectList(ompObjectList, converter, lastPrivateClauseOperands);
+    }
+  }
+  // FIXME: Can be done in a better way ?
+  auto &doConstructEval =
+      eval.getFirstNestedEvaluation().getFirstNestedEvaluation();
+  auto *doStmt = doConstructEval.getIf<Fortran::parser::NonLabelDoStmt>();
+
+  const auto &loopControl =
+      std::get<std::optional<Fortran::parser::LoopControl>>(doStmt->t);
+  if (const auto *bounds =
+          std::get_if<Fortran::parser::LoopControl::Bounds>(&loopControl->u)) {
+    Fortran::lower::StatementContext stmtCtx;
+    lowerBound.push_back(fir::getBase(converter.genExprValue(
+        *Fortran::semantics::GetExpr(bounds->lower), stmtCtx)));
+    upperBound.push_back(fir::getBase(converter.genExprValue(
+        *Fortran::semantics::GetExpr(bounds->upper), stmtCtx)));
+    if (bounds->step) {
+      step.push_back(fir::getBase(converter.genExprValue(
+          *Fortran::semantics::GetExpr(bounds->step), stmtCtx)));
+    }
+    // If `step` is not present, assume it as `1`.
+    else {
+      step.push_back(firOpBuilder.createIntegerConstant(
+          currentLocation, firOpBuilder.getIndexType(), 1));
+    }
+  }
+  // FIXME: Add support for following clauses:
+  // 1. linear
+  // 2. order
+  auto wsLoopOp = firOpBuilder.create<mlir::omp::WsLoopOp>(
+      currentLocation, lowerBound, upperBound, step, privateClauseOperands,
+      firstPrivateClauseOperands, lastPrivateClauseOperands, linearVars,
+      linearStepVars, scheduleClauseOperand.dyn_cast_or_null<StringAttr>(),
+      scheduleChunkClauseOperand,
+      collapseClauseOperand.dyn_cast_or_null<IntegerAttr>(),
+      noWaitClauseOperand.dyn_cast_or_null<UnitAttr>(),
+      orderedClauseOperand.dyn_cast_or_null<IntegerAttr>(),
+      orderClauseOperand.dyn_cast_or_null<StringAttr>());
+
+  // Handle attribute based clauses.
+  for (const auto &clause : wsLoopOpClauseList.v) {
+    if (const auto &collapseClause =
+            std::get_if<Fortran::parser::OmpClause::Collapse>(&clause.u)) {
+      const auto *expr = Fortran::semantics::GetExpr(collapseClause->v);
+      const auto collapseValue = Fortran::evaluate::ToInt64(*expr);
+      wsLoopOp.collapse_valAttr(firOpBuilder.getI64IntegerAttr(*collapseValue));
+    } else if (const auto &orderedClause =
+                   std::get_if<Fortran::parser::OmpClause::Ordered>(
+                       &clause.u)) {
+      const auto *expr = Fortran::semantics::GetExpr(orderedClause->v);
+      const auto orderedValue = Fortran::evaluate::ToInt64(*expr);
+      wsLoopOp.ordered_valAttr(firOpBuilder.getI64IntegerAttr(*orderedValue));
+    } else if (const auto &scheduleClause =
+                   std::get_if<Fortran::parser::OmpScheduleClause>(&clause.u)) {
+      const auto scheduleKind =
+          std::get<Fortran::parser::OmpScheduleClause::ScheduleType>(
+              scheduleClause->t);
+      switch (scheduleKind) {
+      default:
```

Comment:
Do you cover all cases in this switch? If so, the default should not be present. 
https://llvm.org/docs/CodingStandards.html#don-t-use-default-labels-in-fully-covered-switches-over-enumerations